htp/htp_response_generic.c

```diff
@@ -217,8 +219,52 @@ htp_status_t htp_parse_response_header_generic(htp_connp_t *connp, htp_header_t
     }
 
     // Now extract the name and the value.
-    h->name = bstr_dup_mem(data + name_start, name_end - name_start);
-    h->value = bstr_dup_mem(data + value_start, value_end - value_start);
+    // Check if there are null characters inside and remove them
+    nbnull = 0;
+    size_t j = 0;
+    for (i = name_start; i < name_end; i++) {
+        if (data[i] == 0) {
+            nbnull++;
+        }
+    }
+    if (nbnull == 0) {
+        // simple case
+        h->name = bstr_dup_mem(data + name_start, name_end - name_start);
+    } else {
+        htp_log(connp, HTP_LOG_MARK, HTP_LOG_WARNING, 0,
+                "Null character in header line");
+        h->name = bstr_alloc(name_end - name_start - nbnull);
```

Comment:
missing error check?

Reply:
Ok, maybe there is a better code style for this whole thing...